pkg/jobs/traffic/traffic.go

```diff
@@ -0,0 +1,134 @@
+package traffic
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+
+	"github.com/go-co-op/gocron"
+	"golang.org/x/exp/slices"
+
+	"github.com/adh-partnership/api/pkg/config"
+	"github.com/adh-partnership/api/pkg/database"
+	"github.com/adh-partnership/api/pkg/database/models"
+	"github.com/adh-partnership/api/pkg/discord"
+	"github.com/adh-partnership/api/pkg/logger"
+	"github.com/adh-partnership/api/pkg/network/vatsim"
+)
+
+var log = logger.Logger.WithField("component", "job/traffic")
+
+// Max distance for a pilot to be "counted" at an airport.
+const maxDistance = 5.0
+
+// Register the repeating job.
+func ScheduleJobs(s *gocron.Scheduler) error {
+	_, err := s.Every(15).Minutes().SingletonMode().Do(handleTraffic)
+	if err != nil {
+		return fmt.Errorf("failed to schedule traffic job: %s", err)
+	}
+	return nil
+}
+
+// Checks for unstaffed airports that have pilot traffic.
+func handleTraffic() {
+	cfgEntries := config.Cfg.Facility.TrafficAlerts.Entries
+	if len(cfgEntries) == 0 {
+		return
+	}
+	log.Info("Checking traffic for unstaffed airports")
+	airports := []models.Airport{}
+	if err := database.DB.Where("twr_type_code != ?", "NON-ATCT").Find(&airports).Error; err != nil {
+		log.Errorf("Failed to get airports: %s", err)
+		return
+	}
+	networkData, err := vatsim.GetData()
+	if err != nil {
+		log.Errorf("Failed to get network data: %s", err)
+		return
+	}
+
+	needCoverage := make(map[string]uint)
+	for _, airport := range airports {
+		icao, count := checkAirport(cfgEntries, &airport, networkData)
+		if count > 0 {
+			needCoverage[icao] = count
+		}
+	}
+
+	err = sendNotification(needCoverage)
+	if err != nil {
+		log.Errorf("Error sending uncovered traffic message to Discord: %s", err.Error())
+	}
+}
+
+// Checks a single airport for pilot traffic and no top-down coverage, as configured.
+func checkAirport(
+	cfgEntries []config.ConfigFacilityAlertsEntry,
+	airport *models.Airport,
+	networkData *vatsim.VATSIMData,
+) (string, uint) {
+	entryIdx := slices.IndexFunc(cfgEntries, func(e config.ConfigFacilityAlertsEntry) bool {
```

Comment:
If you'd rather not use this library, I can rewrite to use [`sort::Search`](https://pkg.go.dev/sort#Search).